### PR TITLE
docs: add core: labels to issue labeling guide

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Add the new `core:` labels to the issue labeling guide

## Context:

Based on the discussion #10347.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/issue-labeling.md [hunk 1]
```diff
@@ -3,6 +3,17 @@
...
+The general idea is that we have:
+
+- manager (`manager:`)
+- versioning (`versioning:`)
+- datasource (`datasource:`)
+- platform (`platform:`)
+- core functionality (`core:`)
+
+The majority of issues should have at least one of those labels.
+These labels should also map approximately to our semantic commit scopes.
```
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:15]: This is adapted from the explanation you gave in #10347.

File: docs/development/issue-labeling.md [hunk 2]
```diff
@@ -92,6 +103,20 @@ Use [this search](https://github.com/renovatebot/renovate/issues?q=is%3Aissue+is
...
+    <summary>Core labels</summary>
+
+    core:automerge
+    core:dashboard
+    core:onboarding
+    core:schedule
+
+</details>
+
+Use these to .... (@rarkins, please write out how we should use these labels)
```
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:118]: @rarkins can you explain how you want us to use the `core:` label?

**Review by HonkingGoose — COMMENTED** [review]
It's not ready for merge yet, as we need to explain when to apply `core:` labels properly still.

File: docs/development/issue-labeling.md (hunk 1 above)
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:15]:
```suggestion
These labels should also map approximately to our Conventional Commit scopes.
```

I think we're following the Conventional Commits rules?

**Review by HonkingGoose — COMMENTED** [review]
Looks good to go! Just one thing to check. 😉

**Review by rarkins — APPROVED** [review]

**Pull request merged**